File: problem/0965_Univalued Binary Tree.js
// A binary tree is uni-valued if every node in the tree has the same value.

// Given the root of a binary tree, return true if the given tree is uni-valued, or false otherwise.

/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
var isUnivalTree = function(root) {
    const value = root.val;
    return sameVal(root, value)
};

function sameVal(node, value){
    if(!node) return true;
    if (node.val !== value) return false;
    return sameVal(node.left, value) && sameVal(node.right, value)
}

File: c#-solution/0965. Univalued Binary Tree.cs
// 0965. Univalued Binary Tree
// https://leetcode.com/problems/univalued-binary-tree/
// Tag: C#, Tree, Depth-first Search, Binary Tree, Easy, DFS
// Related: 100. Same Tree, 101. Symmetric Tree, 226. Invert Binary Tree

/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public bool IsUnivalTree(TreeNode root) {
        bool isSam(TreeNode node, int val)
        {
            if(node == null) return true;
            if(node.val != val) return false;
            return isSam(node.left, val) && isSam(node.right, val);
        }
        return isSam(root, root.val);
    }
}